library/blivet.py

```diff
@@ -124,6 +124,22 @@
 packages_only = None   # only set things up enough to get a list of required packages
 
 
+def find_duplicate_names(dicts):
+    """ Return a list of names that appear more than once in a list of dicts.
+
+        Items can be a list of any dicts with a 'name' key; that's all we're
+        looking at. """
+    names = list()
```

Comment:
would be faster to use `names = set()` since you only use `names` to see if this is a duplicate or not - then `names.add(item['name'])` instead of `names.append(item['name'])`

Reply:
That's true. I don't think it will matter in practice since it's unlikely users will have more than 10s of pools or volumes at any given level. It would also make sense for `duplicates` to be a set, but I think the randomized ordering is probably worse than the sub-optimal efficiency (and sorting or casting to a list would presumable wash the performance gains of using a set in the first place. So, while I don't mind making the change, I am doubtful it will have any real-world effect.

Reply:
ok - then leave it as is - I thought `duplicates` was a list so that you could show the user how many duplicates there were e.g. the output would be `volA volA volA volB volB` or such, which might be more useful than just `volA volB`, and doing something like `volA:3 volB:2` e.g. show the count of each one is also probably a bit much for 10s of items.

Reply:
Hah! Now you caught a bug. My intention was to only have each name appear once in `duplicates`, but I didn't write it that way. I'm going to fix that.